earth_moving/clutter/engine.py

```diff
@@ -120,23 +124,28 @@ def control_joint(self, joint_index ,joint_target, control_mode=p.VELOCITY_CONTR
         :param control_mode: Integer, the control mode (p.POSITION_CONTROL or p.VELOCITY_CONTROL).
         """
         
+        # Njoints
+        Njoints = p.getNumJoints(self.ID[1])
+        JointList = range(Njoints)
+        
         # control the joint
         if control_mode == p.POSITION_CONTROL:
-            p.setJointMotorControl2(self.ID[1], \
-            joint_index, \
+            p.setJointMotorControlArray(self.ID[1], \
+            JointList, \
             control_mode, \
-            targetPosition=joint_target, \
-            targetVelocity=target_velocity, \
-            maxVelocity=100*target_velocity, \
-            force=max_force, \
-            positionGain=0.0001, \
-            velocityGain=0.0005)
+            targetPositions=joint_target, \
+            targetVelocities=self.Target_velocities, \
+            forces=self.maxForce, \
+            # positionGains=self.Pos_gains, \
+            # velocityGains=self.Vel_gains
+            )
+            # if you want you can play also with 
         elif control_mode == p.VELOCITY_CONTROL:
-            p.setJointMotorControl2(self.ID[1], \
-            joint_index, \
+            p.setJointMotorControlArray(self.ID[1], \
+            JointList, \
             control_mode, \
-            targetVelocity=joint_target, \
-            force=max_force)   
+            targetVelocities=joint_target, \
+            forces=self.maxForce)
             
     # compute the error of the joint with respect to a target
     def compute_joint_error(self, joint_index, target):
```

Comment:
I wonder about this method.. is it indeed part of the Engine class. I'm also not sure it's part of the control class.
This sounds to me more like a utility function, which should lay under a `utils` package. The main different that I see between utility functions and class member functions is their relation to the class (or the lack of it).
For example, if you have a function for calculating Euclidean distance. This is clearly a utility function that should be entangled with any class (and can be called from several class and therefore reused).

Reply:
I don't know, here the <setJointMotorControl2> is the pybullet method to simultaneously control all the joints. So maybe it should be in the same structure of the <control_mode> parameter, with a higher abstraction layer directly sending the command compliant to pybullet